src/components/views/rooms/RoomListPanel/RoomListPrimaryFilters.tsx

```diff
@@ -23,23 +27,181 @@ interface RoomListPrimaryFiltersProps {
+                            >
+                                <ChatFilter
+                                    selected={filter.active}
+                                    onClick={() => {
+                                        onFilterChange();
+                                        filter.toggle();
+                                    }}
+                                >
+                                    {filter.name}
+                                </ChatFilter>
+                            </li>
+                        ))}
+                    </Flex>
+                    {displayChevron && (
+                        <IconButton
+                            aria-expanded={isSafeExpanded}
+                            aria-controls={id}
+                            className="mx_RoomListPrimaryFilters_IconButton"
+                            aria-label={
+                                isSafeExpanded ? _t("room_list|collapse_filters") : _t("room_list|expand_filters")
+                            }
+                            size="28px"
+                            onClick={() => setIsExpanded((_expanded) => !_expanded)}
+                        >
+                            <ChevronDownIcon color="var(--cpd-color-icon-secondary)" />
+                        </IconButton>
+                    )}
+                </Flex>
+            </div>
+        </div>
     );
 }
+
+/**
+ * A hook to sort the filters by active state.
+ * The list is sorted if the current filter is not visible when the list is unexpanded.
+ *
+ * @param filters - the list of filters to sort.
+ * @param isExpanded - the filter is expanded or not (fully visible).
+ * @param isVisible - `null` if there is not selected filter. `true` or `false` if the filter is visible or not.
+ */
+function useFilters(
+    filters: RoomListViewState["primaryFilters"],
+    isExpanded: boolean,
+    isVisible: boolean | null,
+): {
+    /**
+     * The new list of filters.
+     */
+    filters: RoomListViewState["primaryFilters"];
+    /**
+     * Reset the filter sorting when called.
+     */
+    onFilterChange: () => void;
+} {
+    // By default, the filters are not sorted
+    const [filterState, setFilterState] = useState({ filters, isSorted: false });
+
+    useEffect(() => {
+        // If there is no current filter (isVisible is null)
+        // or if the filter list is fully visible (isExpanded is true)
+        // or if the current filter is visible and the list isn't sorted
+        // then we don't need to sort the filters
+        if (isVisible === null || isExpanded || (isVisible && !filterState.isSorted)) {
+            setFilterState({ filters, isSorted: false });
+            return;
+        }
+
+        // Sort the filters with the current filter at first position
+        setFilterState({
+            filters: filters
+                .slice()
+                .sort((filterA, filterB) => (filterA.active === filterB.active ? 0 : filterA.active ? -1 : 1)),
```

Comment:
Can we replace this chained ternary with an if-else?

Reply:
https://github.com/element-hq/element-web/pull/29992/commits/7679b7c4b89bf9baeeddc3f24c0b0926e118cc8c